internal/store/engine_state.go

```diff
@@ -667,6 +667,32 @@ func (ms *ManifestState) HasPendingChangesBeforeOrEqual(highWaterMark time.Time)
 	return ok, earliest
 }
 
+func (ms *ManifestState) UpdateStatus(triggerMode model.TriggerMode) model.UpdateStatus {
```

Comment:
the whole thrust of this PR is just taking this 'if' branch from the client and moving it to the server, and improving it a bit with stuff the server knows (like whether a local resource has an update command)